### ISSUE TITLE
FN question_mark:  if let binding

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5260]
````rust
if let Some(n) = n.try_eval_usize(self.lcx.tcx, self.lcx.param_env) {
    n
} else {
    return None;
}
````
can be simplified to
````rust
n.try_eval_usize(self.lcx.tcx, self.lcx.param_env)?
````
but the lint does not report it
